test/unit/setting_registry_branding_test.rb

```diff
@@ -27,14 +27,16 @@ class SettingRegistryBrandingTest < ActiveSupport::TestCase
     end
 
     test 'hides upstream-only settings' do
-      assert_nil Setting['allow_multiple_content_views']
+      UpstreamOnlySettings.expects(:include?).with('test_setting').returns(true)
+      assert_nil Setting['test_setting']
```

Comment:
You never add `test_setting` to `UpstreamOnlySettings::SETTINGS`, so this test is not testing what it says to be testing.
Wouldn't you need to first declare that setting with a default value, then add it to UpstreamOnlySettings and then assert the result is `nil`?

Reply:
I can't
```
UpstreamOnlySettings::SETTINGS << 'test_setting'
```

due to
```
FrozenError: can't modify frozen Array: []
```

But I did add the setting definition in the test.
(and did confirm it does fail when I comment out the 
```
UpstreamOnlySettings.expects(:include?).with('test_setting').returns(true)
```
)

does that look okay?